app/models/x509_validator.rb

```diff
@@ -46,8 +46,20 @@ def certificate_key_is_supported(record, x509)
   end
 
   def certificate_is_strong(record, x509)
-    return if x509.public_key.params['n'].num_bits >= 2048
+    if x509.public_key.params['n'].num_bits >= 2048
+      hash_algorithm_is_strong(record, x509)
+    else
+      record.errors.add(:certificate, I18n.t('certificates.errors.small_key'))
+    end
+  end
+
+  def valid_algorithms
+    %w[sha256WithRSAEncryption sha384WithRSAEncryption]
```

Comment:
we should keep 512 as we support it: https://github.com/alphagov/verify-saml-libs/blob/b05c2e8630f8f4416dd5498736978ee0927698cf/saml-lib/src/test/java/uk/gov/ida/saml/security/SignatureValidatorTest.java#L114 (at least for signatures) so you can check the hub's source code to see what is supported. Regardless, we should not support anything below sha256...

Reply:
Settled on 256 and 512 as that is what the hub code seems to support.  Removed 384.